tests/plugins/kernel_install/utils_test.sh

```diff
@@ -48,6 +48,35 @@ function test_ask_yN()
 
   output=$(printf '%s\n' 'N' | ask_yN 'Test message')
   assert_equals_helper 'TEST_MODE' "$LINENO" '0' "$output"
+
+  # Tests with default option selected
+  output=$(printf '%s\n' 'y' | ask_yN 'Test message' 'N')
+  assert_equals_helper 'Test_MODE' "$LINENO" '1' "$output"
+
+  output=$(printf '%s\n' 'Y' | ask_yN 'Test message' 'y')
+  assert_equals_helper 'TEST_MODE' "$LINENO" '1' "$output"
+
+  output=$(printf '%s\n' '' | ask_yN 'Test message' 'y')
+  assert_equals_helper 'TEST_MODE' "$LINENO" '1' "$output"
+
+  output=$(printf '%s\n' 'n' | ask_yN 'Test message' 'Y')
+  assert_equals_helper 'TEST_MODE' "$LINENO" '0' "$output"
+
+  output=$(printf '%s\n' 'N' | ask_yN 'Test message' 'n')
+  assert_equals_helper 'TEST_MODE' "$LINENO" '0' "$output"
+
+  output=$(printf '%s\n' '' | ask_yN 'Test message' 'n')
+  assert_equals_helper 'TEST_MODE' "$LINENO" '0' "$output"
+
+  # Invalid default
+  output=$(printf '%s\n' '' | ask_yN 'Test message' 'lala')
+  assert_equals_helper 'TEST_MODE' "$LINENO" '0' "$output"
```

Comment:
After seeing this test, I believe that you need to document this behavior in the function comment. Explain that anything different from `Y/y`, will be treated as `n`.

Reply:
Well, it depends. In this case, it was given an invalid default option to the function, so it will be used 'n' (just like if it was given nothing). If the user chooses nothing, in this case, will be chosen 'n', but if the function was given 'y' as the default option and the user chooses nothing, it will be chosen 'y'. Is this the behavior you are referring to?

Reply:
I think what you should document is the fact that the function only checks if the`response` is a valid yes, otherwise it is considered as a no.